base/loading.jl

```diff
@@ -985,11 +985,7 @@ function explicit_manifest_entry_path(manifest_file::String, pkg::PkgId, entry::
     end
```

Comment:
This is inconsistent now, since the function is supposed to return the path to the file (due to this branch and the stdlib one), but this PR only changes the stdlib one

Reply:
But `path` here will be the path to the directory, no (`dirname(manifest_file)`)?

Reply:
Ah, is it? I was confused by the documentation implying it is the path to the source code:
https://docs.julialang.org/en/v1/manual/code-loading/#Project-environments

But okay, seems you are right that it is expected to be either the path to the source code exactly or a directory containing the source code:

"If any of these result in success, the path to the source code entry point will be that result or the relative path from that result plus src/X.jl")

Reply:
I'm not sure what I have changed to here is correct actually because I think in implicit envs you can just have the file `Package.jl` without putting it in any folder. So maybe this PR should be closed.